plugins/module_utils/api.py

```diff
@@ -655,8 +655,9 @@ def compare_objects(self, current_object, proposed_object):
                         return False
 
                 # If the lists are of a different length the objects and order of element mismatch
-                if proposed_item != current_item:
-                    return False
+                if not key == 'options':
+                    if proposed_item != current_item:
```

Comment:
Consider simplifying the nested `if` statements into a single conditional expression to enhance code readability and maintain consistency across your conditions. 

Also it will be good to add the one liner comment saying why **option** key is ignored.